scipy/stats/_distn_infrastructure.py

```diff
@@ -1056,10 +1056,9 @@ def stats(self, *args, **kwds):
                     mu2p = self._munp(2, *goodargs)
                     if mu is None:
                         mu = self._munp(1, *goodargs)
-                    mu2 = mu2p - mu * mu
-                    if np.isinf(mu):
-                        # if mean is inf then var is also inf
-                        mu2 = np.inf
+                    # if mean is inf then var is also inf
+                    with np.errstate(invalid='ignore'):
+                        mu2 = np.where(np.isfinite(mu), mu2p - mu**2, np.inf)
```

Comment:
This part isn't clear to me, why is the context manager needed here?

Reply:
```python
>>> a = np.array([np.inf])
>>> a - a
RuntimeWarning: invalid value encountered in subtract
array([nan])
>>> with np.errstate(invalid='ignore'): a-a
array([nan])
```

Reply:
hmm, I think most distributions handle this on their so this should be fine.